lib/puppet/functions/ensure_directory.rb

```diff
@@ -0,0 +1,37 @@
+# Takes a generic 'ensure' parameter (or its boolean equivalent)
+# and converts it to an appropriate value for use with directory
+# declaration.
+Puppet::Functions.create_function(:ensure_directory) do
+  # @param ensure_param Pass ensure value here
+  # @return [String] 'directory' or 'absent'
+  #
+  # @example Calling the function with 'present'
+  #   ensure_directory('present') # returns 'directory'
+  #
+  # @example Calling the function with boolean parameter
+  #   ensure_directory(true) # returns 'directory'
+  #
+  # @example
+  #   class myservice::config (
+  #     Enum[present, absent] $ensure = present,
+  #   ) {
+  #     file { '/etc/myservice':
+  #       ensure => ensure_directory($ensure),
+  #       mode   => '0755',
+  #     }
+  #   }
+  dispatch :ensure do
+    param 'Variant[String, Boolean]', :ensure_param
```

Comment:
If you change the type to `Enum[present, absent]` you would not need to check that in an else since you are guaranteed your method gets called with only one of those two values.